torch/csrc/cuda/nccl.cpp

```diff
@@ -118,10 +120,10 @@ static void _check_inputs(TensorList inputs, TensorList outputs, int input_multi
 
     auto input_device = input.get_device();
     // inputs must be on unique devices
-    if (devices.find(input_device) != devices.end()) {
+    if (devices[input_device]) {
```

Comment:
I don't know of any machines with more than 16 GPUs, but do we want a check  that `input_device` is in-bounds?

Reply:
Actually I think I'll just make this a wrapper around `uint64_t`, so we'll have 64 slots, and I can add assertions there.

Reply:
We aren't. I can change that to `.test` which does the check, or increase the size of a bitset to 64 (it doesn't seem to change anything, see https://godbolt.org/g/Kkh8KP).